pkg/network/tracer/tracer_linux_test.go

```diff
@@ -1669,3 +1683,246 @@ func TestKprobeAttachWithKprobeEvents(t *testing.T) {
 
 	assert.Greater(t, p_tcp_sendmsg, int64(0))
 }
+
+func GetFreePort() (string, error) {
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	if err != nil {
+		return "", err
+	}
+
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return "", err
+	}
+	defer l.Close()
+	return strconv.Itoa(l.Addr().(*net.TCPAddr).Port), nil
+}
+
+func TestProtocolClassification(t *testing.T) {
+	cfg := testConfig()
+	if !classificationSupported(cfg) {
+		t.Skip("Classification is not supported")
+	}
+
+	t.Run("with dnat", func(t *testing.T) {
+		// SetupDNAT sets up a NAT translation from 2.2.2.2 to 1.1.1.1
+		netlink.SetupDNAT(t)
+		testProtocolClassification(t, cfg, "localhost", "2.2.2.2", "1.1.1.1")
+	})
+
+	t.Run("with snat", func(t *testing.T) {
+		// SetupDNAT sets up a NAT translation from 6.6.6.6 to 7.7.7.7
+		netlink.SetupSNAT(t)
+		testProtocolClassification(t, cfg, "6.6.6.6", "127.0.0.1", "127.0.0.1")
+	})
+
+	t.Run("without nat", func(t *testing.T) {
+		testProtocolClassification(t, cfg, "localhost", "127.0.0.1", "127.0.0.1")
+	})
+}
+
+func testProtocolClassification(t *testing.T, cfg *config.Config, clientHost, targetHost, serverHost string) {
+	dialer := &net.Dialer{
+		LocalAddr: &net.TCPAddr{
+			IP:   net.ParseIP(clientHost),
+			Port: 0,
+		},
+	}
+
+	tests := []struct {
```

Comment:
do we want to create a test for multiple protocols on the same connection? (i assume the expected value is we return the first classified protocol)